packages/flutter/lib/src/painting/_network_image_web.dart

```diff
@@ -221,6 +222,19 @@ class NetworkImage
         informationCollector: _imageStreamInformationCollector(key),
       );
     }
+
+    switch (webImgElementStrategy) {
+      case image_provider.WebImgElementStrategy.never:
+        return loadViaDecode();
+      case image_provider.WebImgElementStrategy.always:
+        return loadViaImgElement();
+      case image_provider.WebImgElementStrategy.whenNecessary:
+        try {
+          return loadViaDecode();
+        } catch (e) {
+          return loadViaImgElement();
+        }
```

Comment:
Given the "drawbacks" section, it would be nice to log something here to notify the dev that the <img> workaround was needed. That way the dev can be more mindful of CORS errors, and avoid doing double the work in the future by changing the code to `.always`

Reply:
I kind of get the idea but not completely... Can you provide a suggestion?

Reply:
```dart
} catch (error) {
  print("Warning: Could not load the network image at the following URL: $url");
  print("  The error was: $error");
  print(""); 
  print("  This could indicate that the CORS on the image or server is not configured properly.");
  print("  To load your image, Flutter used a native HTML <img> element to render the image without CORS.");
  print("  Consider instead:
  print("    - Configuring CORS correctly on your server, and pass WebImgElementStrategy.never. See <url to web.dev>");
  print("    - Explicitly setting WebImgElementStrategy.always to save Flutter time deciding which strategy to use");
  print("  For more details on the drawbacks of using an <img> element, see <url to Image.network docs>");
  print("  To suppress this warning in the future, set WebImgElementStrategy.warnOnCors = false");
  return loadViaImgElement();
}
```

With this log statement, I'd recommend: 
- setting the default strategy to `WebImgElementStrategy.whenNecessary`
- setting the default `WebImgElementStrategy.warnOnCors = true`

@jezell @slavap I know you both want to avoid slow or buggy behavior by default, but I'm still really concerned that plenty of devs, especially those that come from the mobile world, won't understand what's happening and might not know to use this option if there is no explanation. I think this gives a nice middle ground where: 

- the default behavior is images mostly "just work"
- developers are warned about potentially slow or buggy behavior
- we give a link to explain CORS (eg, [this one](https://web.dev/articles/cross-origin-resource-sharing) from web.dev)
- we give a link explaining the drawbacks (the `Image.network` docs
- developers are first advised to fix CORS properly
- developers are given a fallback option to use the workaround and disable the warning

Would you agree this is a good compromise?

Reply:
@Levi-Lesches No, I don't think it's not a good compromise to turn it on by default because platform views themselves don't just work. Platform views are a hack with too many downsides to enable by default. I've spent hundreds of hours dealing with fallout from using them. They are by far the most error prone part of Flutter web. We'd be far better off if time if the entire HtmlElementView widget was deleted and people were forced to use the web package to insert html elements in the page when that's what they want (which they can already do today).

Unfortunately, CORS is a part of the web. People need to learn to deal with it, or they need to not ship web apps.

Reply:
@Levi-Lesches while I understand while you are coming from, I think its very important that usability never comes at the expense of stability or reliability. Any feature that breaks other features, or potentially puts the stability of your app at risk shouldn't be the default path.

I think maybe something that would be a middle ground would be to do something like those overflow graphics that appear in dev mode, and maybe render yellow text over a background color that explains that the image was blocked by the CORS policy and writes a warning to the console with a link to the docs about this feature. Then as soon as you encounter it. Providing additional guidance at dev time can be very helpful for new developers.

Reply:
So instead of defaulting to `whenNecessary`, still default to `.never`, but print a warning like the above when CORS fails so that devs have an actionable next step. If that's what you're saying, I agree

Reply:
This is a nice idea, but I think I'll make it a separate PR so that we can focus on changing the default behavior in this PR. Thanks for a suggestion.